src/app/shared/navigation-focus/navigation-focus.ts

```diff
@@ -9,16 +11,34 @@ let lastTimeoutId = -1;
 export class NavigationFocus implements OnInit {
   @HostBinding('tabindex') role = '-1';
 
-  constructor(private el: ElementRef) {}
+  private previousLocation: URL;
+  private currentLocation: URL;
+
+  constructor(private el: ElementRef, private router: Router) {}
 
   ngOnInit() {
-    clearTimeout(lastTimeoutId);
-    // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
-    lastTimeoutId = window.setTimeout(() => this.el.nativeElement.focus({preventScroll: true}),
-      100);
+    this.currentLocation = new URL(window.location.href);
```

Comment:
I'm confused about how this works. Wouldn't the location *have* to be `localhost`/`material.io` at this point? I don't know if you're able to detect "hard" navs with this method. Maybe I'm missing something though.

Reply:
There's already code in the `MaterialDocsApp` that tracks `NavigationEnd` events. Can you maybe add another that's something like:
```
router.events
      .pipe(
                filter((event: Event) => event instanceof NavigationEnd)),
                skip(1),
                take(1),
      ).subscribe(() => globalState.internallyNavigated = true);
```

Reply:
Does the router not fire events when I navigate from e.g. about:blank --> m.a.io/components/autocomplete/overview? I was under the impression that it did, but I admittedly don't know much about the router

Reply:
It does fire events for the initial navigation. about:blank --> m.a.io/components/autocomplete/overview would fire all the navigation events. I don't think I'm answering the question you're asking though... By the time any JS runs on the page, the URL is already m.a.io/components/autocomplete/overview

Reply:
Oh, yeah I see what you're saying. I think for this method to work properly, the previousUrl would have to be globally saved and initially null, then we update it after each navigation. I think Annie is planning to switch this back to using the id property to detect soft navs like you originally suggested anyways